### ISSUE TITLE
[bug]: external key support for group internal and script keys

### ISSUE THREAD

**Issue by jharveyb** [lightninglabs/taproot-assets#883]
### Background

We'd like to support external keys, as in those not derived with tapd and possibly not in the Taproot Assets key family, to be used as an internal key for an asset group or a script key. While working on #866 I found issues with signatures produced when using these keys.

We can derive such a key by calling `DeriveKey` on an LND node:

https://github.com/lightninglabs/taproot-assets/blob/80c413d219b422cfd742923bf9bd16c9e56583c7/itest/mint_fund_seal_test.go#L71

And use it to sign for a key spend path asset group witness:

https://github.com/lightninglabs/taproot-assets/blob/80c413d219b422cfd742923bf9bd16c9e56583c7/itest/mint_fund_seal_test.go#L389

I'm pretty confident that the correct key descriptor is being passed here, given that other signing calls for the other two keys in this tests are successful. The same for marshalling and unmarshalling of the group key request and group virtual tx.

I also confirmed that `DeriveGroupKey` isn't checking that the internal key family is the Taproot Assets family.

The branch for reproducing this issue:

https://github.com/lightninglabs/taproot-assets/tree/mint_seal_batch_external_keys

### Steps to reproduce

`make itest icase=mint_fund_seal_assets`

**Comment by guggero** [comment]
The problem is in [`lndclient`](https://github.com/lightninglabs/lndclient/blob/master/signer_client.go#L210) and how the key descriptor is passed to `lnd`. So I don't think this needs additional changes in `tapd`.

**Comment by jharveyb** [comment]
One issue is using `SignPsbt` over `SignOutputRaw` is that we would then need to 'recover' the key family and index of a pubkey, from `tapd` and not within `lnd` as is being done right now. The most relevant LND RPCs seem to be `ListAccounts` and `ListAddresses`, but I couldn't get the status of keys from random families in that output.

The simpler option to unblock #886 is just blanking pubkeys in key descs before calling for signing.

**Comment by jharveyb** [comment]
Ok, the root cause has been found:

We have two RPCs for deriving keys, `DeriveKey` and `DeriveNextKey`. `tapd` uses `DeriveNextKey` exclusively.

Compared to `DeriveNextKey`, `DeriveKey` does not create and store a P2WPKH address from the derived key.

https://github.com/lightningnetwork/lnd/blob/2089a88f4b5e97a19a9d0a3940222cc90946701a/keychain/btcwallet.go#L164

This P2WPKH address is how a key family and index are fetched when a `KeyDescriptor` passed via `SignOutputRaw` has a pubkey but no key locator info (which is removed during the lndclient call).

https://github.com/lightningnetwork/lnd/blob/2089a88f4b5e97a19a9d0a3940222cc90946701a/lnwallet/rpcwallet/rpcwallet.go#L965

TL;DR without switching to `SignPsbt`, we can only use external keys derived via `DeriveNextKey`. A different signer implementation could have different behavior that avoids this issue.
